cardinal/cardinal.go

```diff
@@ -100,6 +100,9 @@ func EachMessage[In any, Out any](wCtx engine.Context, fn func(message.TxData[In
 	return nil
 }
 
+// RegisterMessage registers a message to the world. Cardinal will automatically set up HTTP routes that map to each
+// registered message. Message URLs are take the form of "group/name". A default group, "game", is used
```

Comment:
We would want to make sure to restrict the name of the group and name to alphanumeric and dash ( - )  to make sure there are no issues with this URL registrations

Reply:
added alphanumeric with dash/underscores enforcement and added test cases. [enforce validity on message name and group, clean up message test file](https://github.com/Argus-Labs/world-engine/pull/697/commits/e98585e404df1a7be2a0be0382223ece33afe38b)

also cleaned up some stuff in that commit.